packages/network-controller/src/NetworkController.ts

```diff
@@ -63,32 +73,178 @@ export type NetworkMetadata = {
+  type: RpcEndpointType.Infura;
+  /**
+   * The URL of the endpoint. Expected to be a sort of template with the text
+   * `{infuraProjectId}`, which will get replaced with the Infura project ID.
+   */
+  url: string;
+};
+
+/**
+ * A custom RPC endpoint is a reference to a user-defined server which fronts an
+ * EVM chain. It may refer to an Infura network, but only by coincidence.
+ */
+export type CustomRpcEndpoint = {
+  /**
+   * The user-facing name of the endpoint.
+   */
+  name: string;
+  /**
+   * The identifier for the network client that has been created for this RPC
+   * endpoint.
+   */
+  networkClientId: string;
+  /**
+   * The type of this endpoint, always "custom".
+   */
+  type: RpcEndpointType.Custom;
+  /**
+   * The URL of the endpoint.
+   */
+  url: string;
+};
+
+/**
+ * An RPC endpoint is a reference to a server which fronts an EVM chain. There
+ * are two varieties of RPC endpoints: Infura and custom.
+ *
+ * @see CustomRpcEndpoint
+ * @see InfuraRpcEndpoint
+ */
+export type RpcEndpoint = InfuraRpcEndpoint | CustomRpcEndpoint;
+
+/**
+ * From a user perspective, a network configuration holds information about a
+ * network that a user can select through the client. A "network" in this sense
+ * can explicitly refer to an EVM chain that the user explicitly adds or doesn't
+ * need to add (because it comes shipped with the client). The properties here
+ * therefore directly map to fields that a user sees and can edit for a network
+ * within the client.
  *
- * @property rpcUrl - RPC target URL.
- * @property chainId - Network ID as per EIP-155
- * @property nickname - Personalized network name.
- * @property ticker - Currency ticker.
- * @property rpcPrefs - Personalized preferences.
+ * Internally, a network configuration represents a single conceptual EVM chain,
+ * which is represented tangibly via multiple RPC endpoints. A "network" is then
+ * something for which a network client object is created automatically or
+ * created on demand when it is added to the client.
  */
 export type NetworkConfiguration = {
-  rpcUrl: string;
+  /**
+   * An optional URL that allows the user to view blocks and transactions that
+   * have occurred on the chain.
+   */
+  blockExplorerUrl?: string;
+  /**
+   * The ID of the chain. Represented in hexadecimal format with a leading "0x"
+   * instead of decimal format so that when viewed out of context it can be
+   * unambiguously interpreted.
+   */
   chainId: Hex;
-  ticker: string;
-  nickname?: string;
-  rpcPrefs?: {
-    blockExplorerUrl: string;
-  };
+  /**
+   * The RPC endpoint URL that all requests will use by default in order to
+   * interact with the chain. The value of this property must match the `url`
+   * property of an entry in `rpcEndpoints`.
+   */
+  defaultRpcEndpointUrl: string;
+  /**
+   * The user-facing nickname assigned to the chain.
+   */
+  name: string;
+  /**
+   * The name of the token that represents the native currency for the chain.
```

Comment:
RE https://github.com/MetaMask/core/pull/4286#discussion_r1612090728: using the comments to provide context from a non-technical POV seems like a good middle ground.
```suggestion
   * The name of the token that represents the native currency for the chain (i.e. the ticker symbol). 
```

Reply:
Sure!